### PR TITLE
Cosmetics & some minor improvements

### PR DESCRIPTION
**Pull request by devrck**
Sorry for the big CR.

Easier to look if you see it in your IDE

### CODE REVIEW

**Comment by devrck** [comment]
The configuration for `buid.xml` and `phpunit.xml.dist` is not correct and I don't know how to verify them because I don't have a lot of experience with them.

File: DependencyInjection/EMAGCacheExtension.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        $container->setParameter('emag.cache.service', $config['provider']);
+
+        $container->setParameter('emag.cache.ignore.namespaces', $config['ignore_namespaces']);
+
+        $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
+        $loader->load('services.yml');
+
+        $this->addClassesToCompile([
+            'eMAG\\CacheBundle\\Annotation\\Cache',
+        ]);
```
Comment (alexbumbacea) [inline DependencyInjection/EMAGCacheExtension.php]: ::class notation?

File: EMAGCacheBundle.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        $container->addCompilerPass(new CacheCompilerPass());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function boot()
+    {
+        $this->autoloader = spl_autoload_register($this->container->get('emag.cache.proxy.config')->getProxyAutoloader());
+    }
```
Comment (alexbumbacea) [inline EMAGCacheBundle.php]: spl_autoload_register return bool. Not a reference to the autoloader function. I don't think the shutdown is really required.

File: LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2016 Alexandru Bumbacea
+
```
Comment (alexbumbacea) [inline LICENSE]: copiright emag it research then... :)

File: ProxyManager/CacheFactory.php [hunk 4]
```diff
@@ -1,49 +1,67 @@
...
+     * @return  object
+     */
     public function generate($class, $arguments = [])
     {
         $proxyClassName = $this->proxyConfig->getClassNameInflector()->getProxyClassName($class, [
             'className' => $class,
             'factory' => ProxyCachingObjectFactory::class,
-            'proxyManagerVersion' => \ProxyManager\Version::getVersion()
+            'proxyManagerVersion' => ProxyVersion::VERSION
         ]);
```
Comment (alexbumbacea) [inline ProxyManager/CacheFactory.php]: leave it as function. If they add something to the function... we might not generate the same names

File: README.md [hunk 5]
```diff
@@ -1,28 +1,115 @@
...
+
+
+```php
+    
+    use eMAG\CacheBundle\Annotation as eMAGAnnotation;
+    
+   /**
+     * @eMAGAnnotation\@Cache(cache="<put your prefix>", [key="<name of argument to include in cache key separated by comma>",  [ttl=600, [reset=true ]]])
+     */
+```
```
Comment (alexbumbacea) [inline README.md]: without @ in from of Cache... just in from of emag...

File: Tests/CacheWarmupProxiesTest.php [hunk 6]
```diff
@@ -79,7 +76,7 @@ public function testClassCreated()
             ->getProxyClassName(CacheableClass::class, [
                 'className' => CacheableClass::class,
                 'factory' => ProxyCachingObjectFactory::class,
-                'proxyManagerVersion' => \ProxyManager\Version::getVersion()
+                'proxyManagerVersion' => Version::VERSION
             ]);
```
Comment (alexbumbacea) [inline Tests/CacheWarmupProxiesTest.php]: leave it s function

File: app/config.yml [hunk 7]
```diff
@@ -1,21 +0,0 @@
...
-
-monolog:
-    handlers:
-        main:
-            type:  test
-            level: info
-
-services:
-   cache_warmer:
-      class: Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerAggregate
```
Comment (alexbumbacea) [inline app/config.yml]: where is the new "config" for tests?

**Comment by skafandri** [comment]
I know is not mentioned by PSR, but is a common practice to have the namespace start with an uppercase. I suggest `Emag` or `EMAG` instead of `eMAG`. Note that this is a PHP namespace, not an organization name.

File: Tests/CacheWarmupProxiesTest.php (hunk 6 above)
Comment (devrck) [inline Tests/CacheWarmupProxiesTest.php]: `\ProxyManager\Version::getVersion()` does not exist as a function.

Comment (alexbumbacea) [inline Tests/CacheWarmupProxiesTest.php]: I'll enforce the version in master.

**Comment by devrck** [comment]
Closing for now, will return with another PR because this one has a lot of changes and are not focused on something specific. 

@skafandri  Regarding the namespace I think `EMAG` because I've also named the bundle `EMAGCacheBundle()`

**Pull request closed without merging**